### PR TITLE
primary-site: add KEDA based autoscaling support to inbox-listener

### PR DESCRIPTION
**Pull request by sofuture**
### Changelog

- Feature: add support for built-in inbox-listener autoscaling, see [documentation](https://docs.foxglove.dev/docs/primary-sites/self-hosting/autoscaling#inbox-listener) for details

### Docs

https://docs.foxglove.dev/docs/primary-sites/self-hosting/autoscaling#inbox-listener

### Description

Autoscaling can now be enabled automatically for inbox-listener. Requires KEDA to be installed as a prerequisite.

<table><tr><th>Before</th><th>After</th></tr><tr><td>

</td><td>

</td></tr></table>

Fixes: FG-8702
Fixes: FG-7045
Fixes: FG-8694

### CODE REVIEW

File: charts/primary-site/templates/deployments/inbox-listener.yaml [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+  pollingInterval: 30
+  jobTargetRef:
+    parallelism: 1
+    activeDeadlineSeconds: 86400
+    {{ include "primary-site.inbox-container" . | nindent 4 }}
+  triggers:
+    - type: metrics-api
+      metadata:
+        targetValue: "2"
+        url: "{{ .Values.globals.foxgloveApiUrl }}/internal/platform/v1/pending-imports-stats"
```
Comment (bryfox) [inline charts/primary-site/templates/deployments/inbox-listener.yaml:30]: Suddenly this internal endpoint doesn't feel as internal. Could this feasibly be read from the site controller at startup? Could the site controller itself proxy these requests to the API?

File: charts/primary-site/values.yaml [hunk 2]
```diff
@@ -97,6 +97,15 @@ inboxListener:
...
       ##   eks.amazonaws.com/role-arn: arn:aws:iam::xxxxxxxxxxxx:role/foxglove-inbox-listener-sa-role
 
+  # To enable the autoscaling built into this chart, you must install KEDA first
+  # helm repo add kedacore https://kedacore.github.io/charts
+  # helm repo update
+  # helm install keda kedacore/keda --namespace keda --create-namespace
+  autoscaling:
+    enabled: false
+    minReplicas: 1
+    maxReplicas: 10
```
Comment (bryfox) [inline charts/primary-site/values.yaml:107]: Are these recommended values? Do we have any way to make recommendations?

**Review by bryfox — COMMENTED** [review]
Should this include autoscaling config for the stream service as well?

File: charts/primary-site/values.yaml (hunk 2 above)
Comment (sofuture) [reply charts/primary-site/values.yaml:107]: It depends on load and capacity, so we'll have to expound on this in docs. `False` and `1` are appropriate defaults in any case.

File: charts/primary-site/templates/deployments/inbox-listener.yaml (hunk 1 above)
Comment (bryfox) [reply charts/primary-site/templates/deployments/inbox-listener.yaml:30]: Maybe another option is to have the site-controller serve its cached metric directly; it's already polling this endpoint in the background and has a server running for liveness checks.

**Comment by sofuture** [comment]
@james-rms this is ready to go, there was no way to improve the KEDA install.

**Review by james-rms — APPROVED** [review]

**Pull request merged**